### DISCUSSION TITLE
Difficulty building the project for the first time.

### DISCUSSION THREAD

**Discussion by dut-a** [AnySoftKeyboard/AnySoftKeyboard#3015] [Q&A]
1.  These setup instructions seem dated:
     (a) [Contributing](https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/master/CONTRIBUTING.md)
     (b) [How to Build AnySoftKeyboard](https://github.com/AnySoftKeyboard/AnySoftKeyboard/wiki/How-to-Build-AnySoftKeyboard)
     
     I tried this: `android update project -p .`
     
     **and I got**:
     
     `The "android" command is deprecated.`...`Invalid or unsupported command "update project -p .`
     
     I went for the suggested replacement _**sdkmanager**_, couldn't get an exact replica of the command and had to run `sdkmanager --update`. Nothing useful happened. :(.
     
 2. On running `gradle build`, I keep getting this error:
 
 `FAILURE: Build completed with 2 failures.`
  `1: Task failed with an exception.`
    `-----------`

> `* What went wrong:
> A problem occurred configuring project ':ime:dictionaries:jnidictionaryv1'.
> com.android.builder.errors.EvalIssueException: NDK from ndk.dir at <NDK_DIR_PATH> had version [14.1.3816874] which disagrees with android.ndkVersion [23.0.7599858]
> `

Any workarounds to get it to build?
___
Shouldn't be a big deal but I am trying to set it up on MacOS Big Sur (v11.6, `AArch64`).

**Comment by asl97** [comment]
Here an idea, download and install Ndk 23.0.7599858?

____

The instruction is outdated, the `CONTRIBUTING.md` still says Java 11 while other places on the issue/pull tracker saying Java 17

https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/3f153540ae81b0c52e4c39007baed219df1c9fd7/CONTRIBUTING.md?plain=1#L4

              Seems like a java version issue. You'll need Java 17 installed and set as the jdk for this project

_Originally posted by @menny in https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/3709#issuecomment-1537649702_
            
____

I just want to make an layout identical to hacker keyboard because [android 13 broke a big key feature](https://github.com/klausw/hackerskeyboard/issues/901) and I wasn't expecting all the trouble I encountered, after downloading over 3gb via slow public wifi and hours of waiting and installing ancient version of software, uninstalling, reinstall the actual required modern version of stuff, it finally compiled

If I can't figure out how to do a standalone package then I guess I will just do what others did and just upload the folder containing the pack & apk folder if I am actually motivated enough to complete it
